src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,57 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static int GROWTH_FACTOR;
+    private T[] stack;
+    private int size;
+    private int capacity;
 
+    public MateStack() {
+        GROWTH_FACTOR = 2;
```

Comment:
Consider making `GROWTH_FACTOR` a final instance variable to ensure it remains constant after initialization. This will prevent accidental changes and ensure that each instance of `MateStack` can have its own growth factor.